plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+   *   The first data structure for the comparison.
+   * @param mixed $b
+   *   The second data structure for the comparison.
+   */
+  public function valueSort($a, $b) {
+    return;
+  }
+
+  /**
+   * Sort the list of variables.
+   *
+   * This data provider does not handle compound sorts; the last sort defined
+   * will be the one to take effect.
+   *
+   * @param array $variables
+   *   An array keyed by variable name, valued by unserialized variable value.
+   *
+   * @throws \RestfulBadRequestException
+   */
+  protected function applyListSort(array &$variables) {
+    $public_fields = $this->getPublicFields();
+
+    // Get the sorting options from the request object.
+    $sorts = $this->parseRequestForListSort();
+
+    $sorts = $sorts ? $sorts : $this->defaultSortInfo();
+
+    foreach ($sorts as $public_field_name => $direction) {
+      if (isset($public_fields[$public_field_name]['property'])) {
+        $property_name = $public_fields[$public_field_name]['property'];
+        if ($property_name == 'name') {
+          uksort($variables, array(__CLASS__, 'nameSort'));
+        }
+        elseif ($property_name == 'value') {
+          uasort($variables, array(__CLASS__, 'valueSort'));
+        }
+      }
+      if ($direction == 'DESC') {
+        $variables = array_reverse($variables, TRUE);
+      }
+    }
+
+    return $variables;
+  }
+
+  /**
+   * Set correct page for the index within the array of variables.
+   *
+   * Determine the page that should be seen. Page 1 is actually index 0.
+   *
+   * @param array $variables
+   *   An array keyed by variable name, valued by unserialized variable value.
+   *
+   * @throws \RestfulBadRequestException
+   */
+  protected function applyListPagination(array &$variables) {
+    list($offset, $range) = $this->parseRequestForListPagination();
+    $variables = array_slice($variables, $offset, $range);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTotalCount() {
+    return array('restful_count' => count($this->getVariablesForList()));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function index() {
+    $variables = $this->getVariablesForList();
+    $return = array();
+    foreach ($variables as $variable => $value) {
+      $return[] = $this->view($variable);
+    }
+
+    return $return;
+  }
+
+  /**
+   * Helper function to feed an array into viewMultiple().
+   *
+   * @param string $name_string
+   *  A string of variable names, separated by commas.
+   */
+  public function viewVariables($name_string) {
+    $names = array_unique(array_filter(explode(',', $name_string)));
+    return $this->viewMultiple($names);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewMultiple(array $names) {
+    if (empty($names)) {
+      return array();
+    }
+    $variables = array_keys($this->getVariablesForList());
+    $output = array();
+
+    // Build output according to filtered/sorted list of variables.
+    foreach ($variables as $variable) {
```

Comment:
That is my attempt at sorting logic; I looped through `$variables` because they were in the correct sort order from `getVariablesForList()`.  So, this depends on whether we want this method to sort according to the plugin definition, or the passed-in `$names`.